src/Schema/AbstractColumnSchema.php

```diff
@@ -241,60 +241,39 @@ public function unsigned(bool $value): void
      */
     protected function typecast(mixed $value): mixed
     {
-        if (
-            $value === ''
-            && !in_array(
-                $this->type,
-                [
-                    SchemaInterface::TYPE_TEXT,
-                    SchemaInterface::TYPE_STRING,
-                    SchemaInterface::TYPE_BINARY,
-                    SchemaInterface::TYPE_CHAR,
-                ],
-                true
-            )
-        ) {
-            return null;
-        }
-
-        if (
-            $value === null
-            || $value instanceof ExpressionInterface
-            || gettype($value) === $this->phpType
-        ) {
-            return $value;
-        }
-
-        switch ($this->phpType) {
-            case SchemaInterface::PHP_TYPE_RESOURCE:
-            case SchemaInterface::PHP_TYPE_STRING:
-                if (is_resource($value)) {
-                    return $value;
-                }
-
-                if (is_float($value)) {
-                    /** ensure type cast always has . as decimal separator in all locales */
-                    return DbStringHelper::normalizeFloat($value);
-                }
-
-                if (is_bool($value)) {
-                    return $value ? '1' : '0';
-                }
-
-                return (string) $value;
-            case SchemaInterface::PHP_TYPE_INTEGER:
-                return (int) $value;
-            case SchemaInterface::PHP_TYPE_BOOLEAN:
-                /**
-                 * Treating a 0-bit value as false too.
-                 *
-                 * @link https://github.com/yiisoft/yii2/issues/9006
-                 */
-                return (bool) $value && $value !== "\0";
-            case SchemaInterface::PHP_TYPE_DOUBLE:
-                return (float) $value;
-        }
-
-        return $value;
+        return match (true) {
```

Comment:
If you want to reduce cognitive complexity - don't use match(true).
Do refactoring with remove switch/case and simplify conditions in "if".
Early exit work fastest.

Reply:
Done